addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -245,8 +253,13 @@ def _create_company_from_iap(self, domain):
                     if state:
                         new_company_info['state_id'] = state.id
 
-        new_company_info['iap_enrich_info'] = json.dumps(iap_data)
         new_company = request.env['res.partner'].create(new_company_info)
+        request.env['res.partner.iap'].sudo().create({
```

Comment:
Why not creating with "iap_enrich_info" in values like done before, as the inverse should handle it ?